### DISCUSSION TITLE
How can I apply NBT data to an Entity / LivingEntity?

### DISCUSSION THREAD

**Discussion by ryleu** [Minestom/Minestom#1062] [Help] [answered]
I'm trying to re-create `/summon`. I have my syntax:
```kt
addSyntax({ sender, context ->
    val position = context[positionArgument].fromSender(sender)
    val nbt = context[nbtArgument]
    val instance = context[instanceArgument].instance

    val entity = LivingEntity(context[entityArgument])
    entity.applyNbtSomehow(nbt) // ??? 

    entity.setInstance(instance, position)
}, entityArgument, positionArgument)
```
How can I take `nbt` and slap it into `entity`?

**Comment by GoldenStack** [comment]
`Entity` implements `Taggable` - you can use those methods. If you want to get the actual `TagHandler`, there's `Taggable#tagHandler()`
You might be looking for something like `entity.tagHandler().updateContent(nbt);`

**Reply by ryleu** [reply]
With `ArgumentType.NbtCompound`, it gets mad at me when I try to give it NBT. Am I using the wrong type?

![image](https://user-images.githubusercontent.com/69326171/168174382-6b26c34e-b76b-49e7-a4fb-4532db4f62c1.png)

**Comment by ryleu** [comment, marked as answer]
Turns out I'd have to write my own deserializer. As of writing, Minestom is not data-driven.
